AllReadyApp/Web-App/AllReady.UnitTest/Areas/Admin/Controllers/SiteAdminControllerShould.cs

```diff
@@ -223,11 +294,36 @@ public void EditUserPostReturnsSameViewAndViewModelWhenModelStateIsInvalid()
             await TaskFromResultZero;
         }
 
-        [Fact(Skip = "NotImplemented")]
+        [Fact]
         public async Task EditUserPostSendsSendAccountApprovalEmailWithCorrectDataWhenModelsIsOrganizationAdminIsTrueAndOrganizationAdminClaimWasAddedSuccessfully()
         {
-            //delete this line when starting work on this unit test
-            await TaskFromResultZero;
+            var mediator = new Mock<IMediator>();
+            var userManager = CreateApplicationUserMock();
+            var logger = new Mock<ILogger<SiteController>>();
+            EditUserModel model = new EditUserModel()
+            {
+                IsOrganizationAdmin = true,
+                UserId = It.IsAny<string>()
+
+            };
+            var user = new ApplicationUser()
+            {
+                Id = model.UserId,
+                Email = "test@testy.com"
+            };
+
+            mediator.Setup(x => x.Send(It.Is<UserByUserIdQuery>(q => q.UserId == model.UserId)))
+                .Returns(user);
+            userManager.Setup(x => x.AddClaimAsync(It.IsAny<ApplicationUser>(), It.IsAny<Claim>()))
+                .Returns(() => Task.FromResult(IdentityResult.Success));
+
+            var controller = new SiteController(userManager.Object, logger.Object, mediator.Object);
+            controller.SetDefaultHttpContext();
+            controller.Url = CreateUrlHelperMockObject().Object;
+            await controller.EditUser(model);
+
+            mediator.Verify(m => m.SendAsync(It.Is<SendAccountApprovalEmail>(q => q.Email == user.Email)), Times.Once);
```

Comment:
we want to verify that the SendAccountApprovalEmail message is being sent with the correct data. You test covers the Email property, but does not cover the CallbackUrl property being set. you can setup the mock of IUrlHelper returned from the CreateUrlHelperMockObject, method, and then assert the callback url is equal to the data you setup on the UrlHelperMock